session/session.go

```diff
@@ -326,6 +326,12 @@ func (s *session) doCommit(ctx context.Context) error {
 	//	return kv.ErrRetryable
 	// }
 
+	// mockCommitError8942 is used for PR #8942.
+	// gofail: var mockCommitError8942 bool
+	// if mockCommitError8942 {
+	//	return kv.ErrRetryable
```

Comment:
This mock is incorrect

1. "Transaction too large" is a retryable error ?
2. Is there any clean up after this error is throw?
You know, `s.txn.Commit` does some cleanup if it return error.